src/instructlab/cli/model/upload.py

```diff
@@ -0,0 +1,245 @@
+                fg="red",
+            )
+            raise click.exceptions.Exit(1)
+        click.secho(
+            "\nUploading safetensors model succeeded!",
+            fg="green",
+        )
+
+
+@click.command()
+@click.option(
+    "--model",
+    "-m",
+    required=True,
+    help="Local model to upload.",
+)
+@click.option(
+    "--model-dir",
+    default=lambda: DEFAULTS.CHECKPOINTS_DIR,
+    show_default="The default system model location store, located in the data directory.",
+    help="The local directory to upload the model files from.",
+)
+@click.option(
+    "--dest-type",
+    type=click.Choice(["hf", "oci", "s3"]),
+    required=True,
+    help="The type of destination to upload to - can be 'hf', 'oci', or 's3'",
+)
+@click.option(
+    "--upload-dest",
+    type=str,
+    required=True,
+    help="Destination for the model to be uploaded to. Ex: for a Hugging Face repo, should be '<username>/<repo-name>'",
+)
+@click.option(
+    "--release",
+    "-rl",
+    default="main",
+    show_default=True,
+    help="The revision to upload the model to - e.g. a branch for Hugging Face repositories.",
+)
+@click.option(
+    "--hf-token",
+    default="",
+    envvar="HF_TOKEN",
+    help="User access token for connecting to the Hugging Face Hub.",
+)
+@click.pass_context
+@clickext.display_params
+def upload(ctx, model, model_dir, dest_type, upload_dest, release, hf_token):
+    """Uploads model to a specified location"""
+    uploader = None
+
+    if dest_type == "hf":
+        uploader = HFModelUploader(
+            ctx=ctx,
+            model=model,
+            model_dir=model_dir,
+            upload_dest=upload_dest,
+            release=release,
+            hf_token=hf_token,
+        )
+    elif dest_type == "oci":
+        # TODO: update this when OCI uploading is supported
+        click.secho(
+            f"Uploading of type {dest_type} is not yet supported",
+            fg="yellow",
+        )
+        raise click.exceptions.Exit(0)
+    elif dest_type == "s3":
+        # TODO: update this when S3 uploading is supported
+        click.secho(
+            f"Uploading of type {dest_type} is not yet supported",
+            fg="yellow",
+        )
+        raise click.exceptions.Exit(0)
+    else:
+        click.secho(
+            f"{dest_type} matches neither Hugging Face, OCI registry, or an S3-compatable format.\nPlease supply a supported dest_type",
+            fg="red",
+        )
+        raise click.exceptions.Exit(1)
+
+    try:
+        uploader.upload()
+    # pylint: disable=broad-exception-caught
+    except (ValueError, Exception) as exc:
+        if isinstance(exc, ValueError) and "HF_TOKEN" in str(exc):
+            click.secho(
+                "Uploading to Hugging Face requires a HF Token to be set.\nPlease use '--hf-token' or 'export HF_TOKEN' to upload all necessary models.",
+                fg="yellow",
+            )
+            raise click.exceptions.Exit(0)
```

Comment:
Why exit 0 here?

Reply:
Same answer as above

Reply:
Will change this as well based on the above convo